### PR TITLE
Add Support for aliBuild

### PR DESCRIPTION
**Pull request by PMunkes**
This Pull Request is mainly to keep track of the work necessary to merge FairShip-Support into FairShip.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -141,13 +136,19 @@ find_package(HEPMC)
...
-    set(GSL_DIR ${SIMPATH})
-  Else(${ROOT_LIBRARY_DIR} MATCHES /lib/root)
-    set(BOOST_ROOT ${SIMPATH}/basics/boost)
-    set(GSL_DIR ${SIMPATH}/basics/gsl)
-  EndIf(${ROOT_LIBRARY_DIR} MATCHES /lib/root)
+  IF(DEFINED ENV{SIMPATH})
+    If(${ROOT_LIBRARY_DIR} MATCHES /lib/root)
+      set(BOOST_ROOT ${SIMPATH})
+      set(GSL_DIR ${SIMPATH})
+    Else(${ROOT_LIBRARY_DIR} MATCHES /lib/root)
```
Comment (olantwin) [inline CMakeLists.txt]: Can this case be reached? It seems not.
In that case we should probably remove it.

**Comment by PMunkes** [comment]
TODO: 
- [ ] Transfer all repositories to ShipSoft.
- [ ] Merge all commits from PMunkes/shipdist into aliBuild.
- [x] Upstream changes into FairRoot

File: CMakeLists.txt [hunk 2]
```diff
@@ -304,19 +307,21 @@ MESSAGE(STATUS "Looking for GENIE ...")
 # genie in general
 FIND_PATH(GENIE  NAMES genie PATHS
   ${SIMPATH}/../FairSoft/generators
+  ${GENIE_ROOT}
   NO_DEFAULT_PATH
-)
+  )
```
Comment (olantwin) [inline CMakeLists.txt]: I don't think we need to change the indentation here.

File: field/ShipFieldCreator.cxx [hunk 3]
```diff
@@ -50,7 +50,7 @@ FairField* ShipFieldCreator::createFairField()
 	// Initialise field
 	if ( fMagneticField ) {
 		fMagneticField->Init();
-		fMagneticField->Print();
+		fMagneticField->Print("");
```
Comment (olantwin) [inline field/ShipFieldCreator.cxx]: Why does this need to change?

File: CMakeLists.txt (hunk 1 above)
Comment (PMunkes) [inline CMakeLists.txt]: I am not sure, but I am keeping it in there, to provide feature parity on FairSoft based installations.

File: field/ShipFieldCreator.cxx (hunk 3 above)
Comment (PMunkes) [inline field/ShipFieldCreator.cxx]: Upstream FairRoot no longer allows calling the Print-function with an empty bracket.

Comment (olantwin) [inline field/ShipFieldCreator.cxx]: OK, good to know.

File: CMakeLists.txt (hunk 1 above)
Comment (olantwin) [inline CMakeLists.txt]: I appreciate that you're trying to change as little as possible, but in this case it seems like a clear bug, unless I completely misunderstand this conditional, which we probably should not carry over.

File: CMakeLists.txt [hunk 4]
```diff
@@ -309,7 +309,7 @@ FIND_PATH(GENIE  NAMES genie PATHS
   ${SIMPATH}/../FairSoft/generators
   ${GENIE_ROOT}
   NO_DEFAULT_PATH
-  )
+)
```
Comment (olantwin) [inline CMakeLists.txt]: You could probably just squash this into the previous commit.

Comment (PMunkes) [inline CMakeLists.txt]: Yes, but this can easily be done during the merge.

File: CMakeLists.txt (hunk 1 above)
Comment (PMunkes) [inline CMakeLists.txt]: I accidentally deleted my comment to this section. Here it is reproduced:
I am not sure, why this is here, but I kept it to provide feature parity on FairSoft based installations.

**Comment by olantwin** [comment]
I think all the changes in here are either already implemented by now or out of date, so I suggest we close this pull request?

**Comment by ThomasRuf** [comment]
not anymore uptodate.

**Pull request closed without merging**